sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -168,6 +168,12 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
       return
     }
     lifecycleState = LifecycleState.STATE_DESTROYED
+    // flush the queued events before destroy to avoid lost telemetry events
+    MapProvider.getEventService(mapInitOptions.resourceOptions.accessToken).flush { expected ->
```

Comment:
Nit: Would it be better instead of getting the event service have a `MapProvider.flushPendingEvents`?